### DISCUSSION TITLE
Risk factor: Github can arbitrarily suspend your projects (as in Tornado Cash)

### DISCUSSION THREAD

**Discussion by honobo** [bisq-network/bisq#6318] [General]
Recently they have suddenly suspended the accounts of Tornado Cash devs on Github. For safety, I'd suggest that you people should not depend too much on Github and prepare Plan B in case you are unreasonably banned.

**Comment by brn004** [comment]
Someone should be able to run a backup script that dumps everything from GitHub API then tars it up and stores it in AWS S3.

**Comment by brn004** [comment]
Actually gitprotect may have a special price for open source projects.

https://gitprotect.io/pricing.html

**Comment by honobo** [comment]
AWS is an extremely centralized solution with obvious IP logging, not exactly privacy-respecting nor censorship-resilient. Reasonable options might include:
-Self-hosting Gitlab
-Self-hosting Gitea
-If not self-hosting, [Codeberg](https://docs.codeberg.org/getting-started/what-is-codeberg/) may be philosophically solid, except it's in EU, which seems to want to ban « unhosted wallets ».

Although, admittedly BTC itself is highly centralized and not-privacy oriented anyway. Good luck!

**Reply by Rudy-MS** [reply]
I cannot agree more with honobo's comment. Putting all our eggs Github and specifically AWS(amazon) is just setting up the project for a single point of failure! 

Please consider using Gitea or any other open source decentralized git project.

**Reply by pazza83** [reply]
Codeberg says it is [considering it's stance on blockchain projects](https://codeberg.org/Codeberg/Community/issues/794).

Following SourceHut's [terms of service update to remove cryptocurrency-related projects](https://sourcehut.org/blog/2022-10-31-tos-update-cryptocurrency/).

**Comment by honobo** [comment]
Digital Asset Anti-Money Laundering Act, rather disturbing, should be carefully analyzed, as it could be a direct attack (practically bans) against software developers and nodes for privacy tools in the US.
https://metager.de/meta/meta.ger3?eingabe=Digital+Asset+Anti-Money+Laundering+Act
I hope this bill won't pass...

**Reply by gnrlus** [reply]
That would be impossible to enforce except for small time.

**Comment by gnrlus** [comment]
"open source decentralized"
For reals.

At first glance, this looks promising.
https://wiki.qortal.org/doku.php?id=github_replacement

There used to be a hitchain.org website for this but now it doesn't load.
I'm not sure this project is active but it's the kind of thing to look into.
https://github.com/hitchainrepo

Please forgive the long tangent below, but it is at least partly related to this post since it's something we have to consider in dealing with code hosts...

Re: https://codeberg.org/Codeberg/Community/issues/794
Banning crypto projects is utterly dumb and I think this issue should not be closed even if it gets a little heated. My response to the issue is as follows...

Hammers can build houses or kill people. Cars can take you places or make get aways from bank robberies. Phones can be used to connect with family or coordinate a terror plot. Computers can be used for office work or seriously attacking any number of people and organizations...

So ban everything and enjoy living under a tree, walking everywhere, and using pen and paper. Oh wait, no, you can kill people with pens. You'll have to just remember everything in your head. Oh wait, people can kill people with their bare hands. We better ban people...

Come on. Stop this insanity (at whoever is taking this naive and fearful position). Society will find creative ways to solve problems...

Centralized, fiat, fractional reserve banking is designed to artificially control economies, which destroys generational progress, leading to all kinds of societal chaos and crime, and decentralized crypto projects are the only way out of that trap toward stable economies and ways of life, so let's not shoot ourselves in the feet, please...

Economies that are determined organically and not controlled by pyramid schemes have the real potential to solve all kinds of societal problems...

If we all start creating decentralized crypto economies they will have resistance to manipulation, thus stabilizing the socioeconomic and geopolitical states of affairs, so that people can better meet basic needs and retain and grow multi-generational wisdom and wealth, and not resort to malice, crime, fraud, etc...

And in case they don't know, this is the true reason cryptocurrency was created. It was a market-driven response to the 2008 financial crisis, because we went from a well to do middle class (much of it business owner class even) in the 1960s, which was about 75% of the US population, who had multiple properties, cars, boats, and sometimes even planes, large diverse investments, large families, and fat retirements-- all the way to a thin veneer of a broke as a joke middle class barely hanging on for dear life, working paycheck to paycheck with no assets, no savings, no kids or very miserable kids, no nice vacays, can't even pay the deductible for healthcare, and forget a comfortable retirement. All that is thanks to the fraud that is the US banking system which steals the wealth we create by printing money infinitely, inflating the dollar, and crashing the economy and buying it back for pennies on the dollar, and repeating this cycle endlessly. We came up with crypto because we're sick of this BS...

Plus, US dollars are already being used for every kind of nefarious purpose and still very much preferred for crime. There are plenty millions of anonymous, off-shore, off-record trust babies with trillions of dollars; just ask a seasoned, senior estate attorney from the Carribean. You're never going to stop crime with prohibition and certainly not by banning crypto projects. Criminals are smart, diligent, and motivated, and there are endless loopholes in any system. You only begin to fix things at the family and community level and that can only happen in stable, not rigged, economies, so that generational wisdom and progress don't get undone so easily, and we do a better job of raising and caring for our people and self-governing! And that's the world we live in, so host them *decentralized* crypto projects and start buying and selling hammers and cars with them (for legitimate purposes of course)!

Can I get a hearty amen out there?

**Comment by honobo** [comment]
The good news is, EFF is not serious about this (TC):
https://www.eff.org/deeplinks/2023/05/court-accepts-effs-amicus-brief-tornado-cash-case-right-publish-code
